metricbeat/module/docker/memory/memory_test.go

```diff
@@ -79,6 +79,9 @@ func TestMemoryService_GetMemoryStats(t *testing.T) {
 		},
 	}
 	expectedFields := common.MapStr{
+		"stats": map[string]uint64{
+			"total_rss": 5,
```

Comment:
Maybe, according to the naming conventions, this field should be called `rss.count` https://www.elastic.co/guide/en/beats/devguide/current/event-conventions.html

Reply:
How much do we want to fiddle with the stuff being scraped from the kernel interface? The full list of `total_*` events is generated programmatically by the kernel.

Reply:
Like, this is what's happening in the kernel:

```
for (i = 0; i < ARRAY_SIZE(memcg1_events); i++)
		seq_printf(m, "total_%s %llu\n", memcg1_event_names[i],
			   (u64)memcg_events(memcg, i));

	for (i = 0; i < NR_LRU_LISTS; i++)
		seq_printf(m, "total_%s %llu\n", mem_cgroup_lru_names[i],
			   (u64)memcg_page_state(memcg, NR_LRU_BASE + i) *
			   PAGE_SIZE);
```
Seems like a potential lost cause to figure out what possible fields we would need to re-parse?

Reply:
having that Docker doesn't document these fields and is just passing them in raw I think that we could fail if we try to convert them. Could you update `data.json`? we should see how this looks by default there.

Reply:
Just another opinion. I think that for this case it'd be fine to just copy the stats map, as they are values taken directly from the OS, and I don't think they are going to change a lot. Also they seem to follow a good enough format (lowercase, snake case...).

But it is true that we risk to have fields explosion or other problems if some fancy feature is merged into the kernel.

Some things that we could do to improve this:
* Apply an schema to these memory stats, so we control what variables we collect, and potentially we adapt them to some other schema as Mario proposed (although I am fine with keeping the naming given by docker/OS)
* Explicitly document the fields we know (even if we leave the wildcard to catch new fields)

And +1 to have a `data.json` to have a better view of how it looks. It could be also nice to try it on Windows, to see if it has its own set of variables.

Reply:
> Apply an schema to these memory stats

I initially thought of that, but schema.Apply seemed a bit like overkill considering everything is already a `uint64`. Should we just not use the type conversion part of it?

> Explicitly document the fields we know (even if we leave the wildcard to catch new fields)

How would that work? Is there some way to document this besides `fields.yml` ?

Reply:
passing this in raw sounds reasonable enough, worst case scenario at some point we discover this generates too much data, which seems unlikely anyway.

Updating `data.json` will help to give a good example of what to expect there

Reply:
> > Explicitly document the fields we know (even if we leave the wildcard to catch new fields)
> 
> How would that work? Is there some way to document this besides `fields.yml` ?

Yeah, I mean to add to `fields.yml` the fields we know, but as they look like quite low level metrics, I am ok too with documenting the `docker.memory.stats` group only.

Reply:
Yah, the fact that they're so low level but also consistently typed makes me not too worried about them. I don't think there's any downside to the wildcard yml entry from the perspective of mapping, is there?